src/main/java/graphql/normalized/ExecutableNormalizedOperationToAstCompiler.java

```diff
@@ -11,36 +12,62 @@
 import graphql.language.NullValue;
 import graphql.language.ObjectField;
 import graphql.language.ObjectValue;
+import graphql.language.OperationDefinition;
 import graphql.language.Selection;
 import graphql.language.SelectionSet;
 import graphql.language.TypeName;
 import graphql.language.Value;
+import graphql.schema.GraphQLSchema;
+import graphql.util.FpKit;
 
 import java.util.ArrayList;
+import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 
 import static graphql.collect.ImmutableKit.map;
 import static graphql.language.Argument.newArgument;
 import static graphql.language.Field.newField;
 import static graphql.language.InlineFragment.newInlineFragment;
+import static graphql.language.OperationDefinition.Operation.MUTATION;
 import static graphql.language.OperationDefinition.Operation.QUERY;
-import static graphql.language.OperationDefinition.newOperationDefinition;
+import static graphql.language.OperationDefinition.Operation.SUBSCRIPTION;
 import static graphql.language.SelectionSet.newSelectionSet;
 import static graphql.language.TypeName.newTypeName;
 
 @Internal
 public class ExecutableNormalizedOperationToAstCompiler {
-
-    public static Document compileToDocument(List<ExecutableNormalizedField> topLevelFields) {
+    public static Document compileToDocument(GraphQLSchema schema, List<ExecutableNormalizedField> topLevelFields) {
```

Comment:
We should not need a schema to print a NormalizedOperation. I think we need to add the operation to the `ExecutableNormalizedOperation` itself or pass it here into the function.